src/main/java/DukeParser.java

```diff
@@ -0,0 +1,106 @@
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class DukeParser {
+    private String filePath;
+
+    public DukeParser(String dukeFilePath) {
+        this.filePath = dukeFilePath;
+    }
+
+    /**
+     * This method copies the contents of the file from filePath and
+     * converts them into Task objects into the taskArrayList with the
+     * help of the parse() method.
+     *
+     * @return taskArrayList the ArrayList to copy the information to
+     * @throws IOException if filePath does not exist
+     */
+    public ArrayList<Task> copyFileContents() throws IOException {
+        File f = new File(this.filePath); // create a File for the given file path
+        Scanner s = new Scanner(f); // create a Scanner using the File as the source
+        StringBuilder targetBuilder = new StringBuilder();
+        while (s.hasNext()) {
+            targetBuilder.append(s.nextLine()).append("\n");
+        }
+
+        return parse(targetBuilder.toString());
+    }
+
+    /**
+     * This method parses the string copied from duke.txt and converts them into task objects
+     * into the taskArrayList.
+     *
+     * @param toParse the string to parse
+     * @return a taskArrayList made up of tasks
+     */
+    public static ArrayList<Task> parse(String toParse) {
```

Comment:
Would it be possible to extract some of the code into another function to reduce code repetition and length of the parse method?